lib/excon/ssl_socket.rb

```diff
@@ -116,20 +116,17 @@ def initialize(data = {})
 
       begin
         if @nonblock
-          loop do
-            begin
-              @socket.connect_nonblock
-              break # connect succeeded
-            rescue OpenSSL::SSL::SSLError => error
-              # would block, rescue and retry as select is non-helpful
-              raise error unless error.message == 'read would block'
-            end
+          begin
+            @socket.connect_nonblock
+          rescue IO::WaitReadable
+            IO.select([@socket])
```

Comment:
`IO.select` automatically calls `to_io`. 

If you select on socket errors, wouldn't that mean you retry forever effectively?